MantidPlot/src/Spectrogram.cpp

```diff
@@ -237,6 +239,25 @@ MantidQt::API::QwtRasterDataMD *Spectrogram::dataFromWorkspace(
   auto dim1 = workspace->getDimension(1);
   Mantid::coord_t minX(dim0->getMinimum()), maxX(dim0->getMaximum()),
       minY(dim1->getMinimum()), maxY(dim1->getMaximum());
+
+  // A MatrixWorkspace can be ragged. Make sure the x axis covers all
+  // histograms.
+  auto matrixWorkspace =
+      boost::dynamic_pointer_cast<const Mantid::API::MatrixWorkspace>(
+          workspace);
+  if (matrixWorkspace) {
+    for (size_t iHisto = 0; iHisto < matrixWorkspace->getNumberHistograms();
+         ++iHisto) {
+      auto &x = matrixWorkspace->x(iHisto);
```

Comment:
`const auto& x`?

Reply:
Should we put this code in https://github.com/mantidproject/mantid/blob/master/Framework/API/src/MatrixWorkspace.cpp#L1357 instead and have `MWXDimension` refer to the X-array with the largest extent?

Reply:
* `x` is `const` because `matrixWorkspace` is `const`. Plus method `x(...)` returns a `const`. Do we need another `const` here?
* I am not sure that this will work. My understanding that a MatrixWorkspace can be an MD workspace only when all x vectors are the same.

Reply:
@mantid-roman The `const` is not strictly necessary here but I think it communicates the intent better.  

I see what you're saying although `getDimension` always worked for `MatrixWorkspace` anyway and has never checked the validity of the common bins. `MWXDimension` has always just arbitrarily referred to `x(0)` but we could just have it refer to he largest X vector and then that will at least ensure the dimension can encompass all of the data.

Reply:
@martyngigg There may not be a single x vector that covers the whole range. The axis vector has to be a combination of different vectors. How to bin it? Are you sure this change will not break the code that uses dimensions elsewhere?

Reply:
@mantid-roman Good point. Let's leave this how it is for the release but I think we should see what effect it has so can we put this change into `master`?